airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -82,7 +119,271 @@ public JsonNode toJdbcConfig(JsonNode mssqlConfig) {
+      // todo: ensure this works for Azure managed SQL (since it uses different sql server agent)
+      checkOperations.add(database -> {
+        try {
+          List<JsonNode> queryResponse = database.query(connection -> {
+            final String sql = "SELECT status_desc FROM sys.dm_server_services WHERE [servicename] LIKE 'SQL Server Agent%'";
+            PreparedStatement ps = connection.prepareStatement(sql);
+            LOGGER.info(String.format("Checking that the SQL Server Agent is running using the query: '%s'", sql));
+            return ps;
+          }, JdbcUtils::rowToJson).collect(toList());
+
+          if ( ! (queryResponse.get(0).get("status_desc").toString().contains("Running"))) {
+            throw new RuntimeException(String.format(
+                "The SQL Server Agent is not running. Current state: '%s'. Please check the documentation on ensuring SQL Server Agent is running.",
+                queryResponse.get(0).get("status_desc").toString()));
+          }
+        } catch (Exception e) {
+          if (e.getCause() != null && e.getCause().getClass().equals(com.microsoft.sqlserver.jdbc.SQLServerException.class)) {
+            LOGGER.warn(String.format("Skipping check for whether the SQL Server Agent is running, SQLServerException thrown: '%s'",
+                e.getMessage()));
+          } else {
+            throw e;
+          }
+        }
+      });
+
+      // check that snapshot isolation is allowed
+      checkOperations.add(database -> {
+        List<JsonNode> queryResponse = database.query(connection -> {
+          final String sql = "SELECT name, snapshot_isolation_state FROM sys.databases WHERE name = ?";
+          PreparedStatement ps = connection.prepareStatement(sql);
+          ps.setString(1, config.get("database").asText());
+          LOGGER.info(String.format("Checking that snapshot isolation is enabled on database '%s' using the query: '%s'",
+              config.get("database").asText(), sql));
+          return ps;
+        }, JdbcUtils::rowToJson).collect(toList());
+
+        if (queryResponse.size() < 1) {
+          throw new RuntimeException(String.format(
+              "Couldn't find '%s' in sys.databases table. Please check the spelling and that the user has relevant permissions (see docs).",
+              config.get("database").asText()));
+        }
+
+        if (queryResponse.get(0).get("snapshot_isolation_state").asInt() != 1) {
+          throw new RuntimeException(String.format(
+              "Detected that snapshot isolation is not enabled for database '%s'. MSSQL CDC relies on snapshot isolation. "
+                  + "Please check the documentation on how to enable snapshot isolation on MS SQL Server.",
+              config.get("database").asText()));
+        }
+      });
+
+    }
+
+    return checkOperations;
+  }
+
+  @Override
+  public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(JsonNode config,
```

Comment:
This would cause compilation error, in master the signature of the method has been changed